src/ProxyClient/Http/HttpAdapter.php

```diff
@@ -254,6 +270,13 @@ private function setBaseUri($uriString = null)
      */
     private function filterUri($uriString, array $allowedParts = [])
     {
+        if (!is_string($uriString)) {
+            throw new \InvalidArgumentException(sprintf(
+                'Uri parameter must be a string, %s given',
+                gettype($uriString)
```

Comment:
Maybe output object instance and/or resource types too? I saw a oneliner for that somewhere